### PR TITLE
Add upload-bundle route

### PR DESCRIPTION
**Pull request by Jasopaum**
Depends on https://github.com/arkhn/pyfhirstore/pull/27

This change is [Reviewable](https://reviewable.io/reviews/arkhn/warehouse-api/43)

### CODE REVIEW

File: fhir-api/api.py [hunk 1]
```diff
@@ -111,6 +116,28 @@ def search(resource_type):
...
+    # TODO methodology to avoid/process huge bundles
+    bundle_data = request.get_json(force=True)
+    if not "resourceType" in bundle_data or bundle_data["resourceType"] != "Bundle":
+        raise OperationOutcome("input must be a FHIR Bundle resource")
+
+    for entry in bundle_data["entry"]:
+        resource = entry["resource"]
+        resource_type = resource["resourceType"]
+        if resource_type not in resources_models:
+            raise OperationOutcome("Unknown resource type")
```
Comment (LaRiffle) [inline fhir-api/api.py]: Add the type found in the msg error?

File: fhir-api/db/__init__.py [hunk 2]
```diff
@@ -58,7 +58,7 @@ def get_store():
     global store
     global cached_resources
     connection = get_db_connection()
-    connection_es = get_es_connection()
+    connection_es = None#get_es_connection()
```
Comment (simonvadee) [inline fhir-api/db/__init__.py]: oopsies

File: fhir-api/api.py [hunk 3]
```diff
@@ -12,6 +15,8 @@
 
 from flask_cors import CORS
 
+from db import get_store
```
Comment (simonvadee) [inline fhir-api/api.py]: this is not used

File: fhir-api/api.py [hunk 4]
```diff
@@ -111,6 +116,28 @@ def search(resource_type):
     return jsonify(results)
 
 
+@api.route("/upload-bundle", methods=["POST"])
+def upload_bundle():
+    # TODO methodology to avoid/process huge bundles
+    bundle_data = request.get_json(force=True)
+    if not "resourceType" in bundle_data or bundle_data["resourceType"] != "Bundle":
```
Comment (simonvadee) [inline fhir-api/api.py]: use fhirstore.upload_bundle instead ?

File: fhir-api/models/resource.py [hunk 5]
```diff
@@ -34,14 +36,12 @@ def create(self):
...
 
         try:
-            if self.db.read(self.resource_type, self.id):
-                raise OperationOutcome(f"Resource {self.id} already exist")
-        except NotFoundError:
-            pass
+            self.resource = self.db.create(
+                {**self.resource, "resourceType": self.resource_type, "id": self.id}
+            )
+        except DuplicateKeyError:
```
Comment (simonvadee) [inline fhir-api/models/resource.py]: much  better

**Pull request merged**